src/Images/SampleDistanceSlider.jsx

```diff
@@ -1,66 +1,73 @@
 import applyContrastSensitiveStyleToElement from '../applyContrastSensitiveStyleToElement'
 import '../style.css'
+import Button from 'react-bootstrap/Button'
+import Form from 'react-bootstrap/Form'
+import Image from 'react-bootstrap/Image'
+import Tooltip from 'react-bootstrap/Tooltip'
+import OverlayTrigger from 'react-bootstrap/OverlayTrigger'
+import cn from 'classnames'
 
-function SampleDistanceSlider(props) {
+const SampleDistanceSlider = React.memo(function SampleDistanceSlider(props) {
   const { service } = props
   const spacingDiv = useRef(null)
   const spacingElement = useRef(null)
-  const [state, send] = useActor(service)
-  const name = state.context.images.selectedName
-  const actorContext = state.context.images.actorContext.get(name)
+  const stateContext = useSelector(service, (state) => state.context)
+  const send = service.send
+  const name = stateContext.images.selectedName
+  const actorContext = stateContext.images.actorContext.get(name)
 
   useEffect(() => {
     applyContrastSensitiveStyleToElement(
-      state.context,
+      stateContext,
       'invertibleButton',
       spacingDiv.current
     )
-    state.context.images.volumeSampleDistanceDiv = spacingDiv.current
-    state.context.images.volumeSampleDistanceSlider = spacingElement.current
+    stateContext.images.volumeSampleDistanceDiv = spacingDiv.current
+    stateContext.images.volumeSampleDistanceSlider = spacingElement.current
   }, [])
 
   const spacingChanged = (val) => {
     send({
       type: 'IMAGE_VOLUME_SAMPLE_DISTANCE_CHANGED',
       data: {
-        name: state.context.images.selectedName,
+        name: stateContext.images.selectedName,
         volumeSampleDistance: val
       }
     })
   }
 
   return (
     <div className="iconWithSlider">
-      <Tooltip
-        ref={spacingDiv}
-        title="Volume sample distance"
-        PopperProps={{
-          anchorEl: spacingDiv.current,
-          disablePortal: true,
-          keepMounted: true
-        }}
+      <OverlayTrigger
+        transition={false}
+        overlay={<Tooltip>Volume sample distance</Tooltip>}
       >
-        <Icon className="sampleDistanceButton" style={{ margin: '0 10px 0 0' }}>
-          <img src={sampleDistanceIconDataUri} />
-        </Icon>
-      </Tooltip>
-      <Slider
-        ref={spacingElement}
-        className="slider"
-        min={0}
-        max={1}
-        value={actorContext.volumeSampleDistance}
-        step={0.01}
-        onChange={(_e, val) => {
-          spacingChanged(val)
-        }}
-      />
+        <div className="icon-button-disabled" ref={spacingDiv}>
+          <Image src={sampleDistanceIconDataUri}></Image>
+        </div>
+      </OverlayTrigger>
+      <Form>
+        <Form.Group>
+          <Form.Control
+            ref={spacingElement}
+            type="range"
+            custom
+            className="slider"
+            min={0}
+            max={1}
+            value={actorContext.volumeSampleDistance}
```

Comment:
When I drag the slider, the sample distance changes, but the tick mark does not move.  Something is up here...

Reply:
What the what!  Some sort of re-render update order thing???  Flow should be `slider event -> send state event -> state system magic -> useSelector triggers re-render with updated volumeSampleDistance`.

Reply:
[Someone had a similar issue](https://stackoverflow.com/questions/64882133/react-memo-not-re-render-if-props-change), but I am not sure if what is happening there applies to our case. I suspect that the issue is in the `spacingChanged` -- it gets that the value is changing as we toggle the slider, but I guess it does not send that change to the parent when I use React.memo?!

In any case, I deleted the React.memo in the next versions for the reasons you pointed out above.

Reply:
OK, I'll update performance issue to track this.

Reply:
Thanks!